wicker/core/storage.py

```diff
@@ -27,7 +29,62 @@
 logger = logging.getLogger(__name__)
 
 
-class S3DataStorage:
+class AbstractDataStorage(ABC):
+    """Abstract data storage class that implements required methods for Column Bytes File Cache"""
+
+    @abstractmethod
+    def fetch_file(self, input_path: str, local_prefix: str, timeout_seconds: int) -> str:
+        """Fetch file from chosen data storage method.
+
+        :param input_path: input file path
+        :param local_prefix: local path that specifies where to download the file
+        :param timeout_seconds: number of seconds till timing out on waiting for the file to be downloaded
+        :return: local path to the downloaded file
+
+        """
+        pass
+
+
+class LocalDataStorage(AbstractDataStorage):
+    """Storage routines for reading and writing objects in local fs"""
+
+    def __init__(self) -> None:
+        """Constructor for a local FS"""
+        pass
+
+    @retry(
+        Exception,
+        tries=get_config().storage_download_config.retries,
+        backoff=get_config().storage_download_config.retry_backoff,
+        delay=get_config().storage_download_config.retry_delay_s,
+        logger=logger,
+    )
+    def download_with_retries(self, input_path: str, local_path: str):
+        try:
+            shutil.copyfile(input_path, local_path)
+        except Exception as e:
+            logging.error(f"Failed to download/move object for file path: {input_path}")
+            raise e
+
```

Comment:
@isaak-willett when we get to the `LocalDataset` class in the next PR, it should have an option to skip fetching files (i.e. just leave the local files in place). Perhaps the `LocalDataset` simply represents either a truly local data or it uses a mounted FUSE system that already has an internally managed cache (this is how GCSFuse with the cache option works).